dnf-behave-tests/features/microdnf/repolist.feature

```diff
@@ -73,3 +73,57 @@ Scenario: Repolist with "--enabled --disabled"
       dnf-ci-thirdparty\s+dnf-ci-thirdparty test repository\s+disabled
       dnf-ci-thirdparty-updates\s+dnf-ci-thirdparty-updates test repository
       """
+
+
+# Tests for "--disablerepo" and "--enablerepo" including wildcards support
+@bz1781420
+Scenario: Disable all repos and then enable "dnf-ci-fedora-updates" repo
+   When I execute microdnf with args "repolist --disablerepo=* --enablerepo=dnf-ci-fedora-updates --all"
+   Then the exit code is 0
+    And stdout matches line by line
+      """
+      repo id\s+repo name
+      dnf-ci-fedora\s+dnf-ci-fedora test repository\s+disabled
+      dnf-ci-fedora-updates\s+dnf-ci-fedora-updates test repository\s+enabled
+      dnf-ci-thirdparty\s+dnf-ci-thirdparty test repository\s+disabled
+      dnf-ci-thirdparty-updates\s+dnf-ci-thirdparty-updates test repository\s+disabled
+      """
+
+
+Scenario: Disable "dnf-ci-thirdparty*" repos and enable "dnf-ci-fedora*" repos
+   When I execute microdnf with args "repolist --disablerepo=dnf-ci-thirdparty* --enablerepo=dnf-ci-fedora* --all"
+   Then the exit code is 0
+    And stdout matches line by line
+      """
+      repo id\s+repo name
+      dnf-ci-fedora\s+dnf-ci-fedora test repository\s+enabled
+      dnf-ci-fedora-updates\s+dnf-ci-fedora-updates test repository\s+enabled
+      dnf-ci-thirdparty\s+dnf-ci-thirdparty test repository\s+disabled
+      dnf-ci-thirdparty-updates\s+dnf-ci-thirdparty-updates test repository\s+disabled
+      """
+
+
+Scenario: Only "*-updates" repos are enabled
+   When I execute microdnf with args "repolist --disablerepo=* --enablerepo=*-updates --all"
+   Then the exit code is 0
+    And stdout matches line by line
+      """
+      repo id\s+repo name
+      dnf-ci-fedora\s+dnf-ci-fedora test repository\s+disabled
+      dnf-ci-fedora-updates\s+dnf-ci-fedora-updates test repository\s+enabled
+      dnf-ci-thirdparty\s+dnf-ci-thirdparty test repository\s+disabled
+      dnf-ci-thirdparty-updates\s+dnf-ci-thirdparty-updates test repository\s+enabled
+      """
+
+
+Scenario: Test '?' wildcard. Only "dnf-ci-fedora-updates" repo is enabled
+   When I execute microdnf with args "repolist --disablerepo=* --enablerepo=dnf-ci-??????-* --all"
+   Then the exit code is 0
+    And stdout matches line by line
+      """
+      repo id\s+repo name
+      dnf-ci-fedora\s+dnf-ci-fedora test repository\s+disabled
+      dnf-ci-fedora-updates\s+dnf-ci-fedora-updates test repository\s+enabled
+      dnf-ci-thirdparty\s+dnf-ci-thirdparty test repository\s+disabled
+      dnf-ci-thirdparty-updates\s+dnf-ci-thirdparty-updates test repository\s+disabled
+      """
```

Comment:
The tests look good. Just one question, why don't you use `stdout is` and list the output with the spaces? No big difference here, but I find it easier to read, this with the `\s+` just blends everything together. And, as of now, you only get a pretty colored diff when there's a mismatch with the `stdout is` step :slightly_smiling_face:

Reply:
@lukash Can you merge please this https://github.com/rpm-software-management/microdnf/pull/66 first . Then microdnf repolist command will be print space characters on the end of lines only if output is a terminal.